eva/third_party/huggingface/create.py

```diff
@@ -0,0 +1,208 @@
+from eva.catalog.models.udf_metadata_catalog import UdfMetadataCatalogEntry
+
+UNSUPPORTED_TASKS = [
+    "feature-extraction",
+    "token-classification",
+    "question-answering",
+    "visual-question-answering",
+    "document-question-answering",
+    "table-question-answering",
+    "conversational",
+    "fill-mask",
+    "zero-shot-classification",
+    "zero-shot-image-classification",
+    "zero-shot-object-detection",
+    "zero-shot-audio-classification",
+]
+
+
+def run_pipe_through_text(pipe: Pipeline):
+    input = "The cat is on the mat"
+    return pipe(input)
+
+
+def run_pipe_through_image(pipe: Pipeline):
+    width, height = 224, 224
+    dummy_image = Image.new("RGB", (width, height), "white")
+    draw = ImageDraw.Draw(dummy_image)
+
+    circle_radius = min(width, height) // 4
+    circle_center = (width // 2, height // 2)
+    circle_bbox = (
+        circle_center[0] - circle_radius,
+        circle_center[1] - circle_radius,
+        circle_center[0] + circle_radius,
+        circle_center[1] + circle_radius,
+    )
+    draw.ellipse(circle_bbox, fill="yellow")
+
+    return pipe(dummy_image)
+
+
+def run_pipe_through_audio(pipe: Pipeline):
+    duration_ms, sample_rate = 1000, 16000
+    num_samples = int(duration_ms * sample_rate / 1000)
+    silent_data = np.random.rand(num_samples)
+    return pipe(silent_data)
+
+
+def run_pipe_through_video(pipe: Pipeline):
+    width, height, fps, duration_sec = 224, 224, 30, 1
+    num_frames = fps * duration_sec
+    blank_frame = np.zeros((height, width, 3), dtype=np.uint8)
+
+    with tempfile.NamedTemporaryFile(delete=False, suffix=".mp4") as temp_file:
+        fourcc = cv2.VideoWriter_fourcc(*"mp4v")
+        video_writer = cv2.VideoWriter(temp_file.name, fourcc, fps, (width, height))
+
+        for _ in range(num_frames):
+            video_writer.write(blank_frame)
+
+        video_writer.release()
+        output = pipe(temp_file.name)
+    return output
+
+
+def run_pipe_through_multi_modal_text_image(pipe: Pipeline):
+    width, height = 224, 224
+    image_input = Image.new("RGB", (width, height), "white")
+    text_input = "This is a dummy text input"
+    return pipe(image=image_input, text=text_input)
+
+
+def infer_output_type(**pipeline_args):
+    """
+    Infer the output type of a pipeline based on the task type.
+    Note: The below code is flaky and depends on the version of transformers.
+    It also hard-codes the type assignment to certain tasks.
+    """
+    if "task" not in pipeline_args:
+        raise Exception("Task Not Found In Model Definition")
+    task = pipeline_args["task"]
+    if task in UNSUPPORTED_TASKS:
+        raise Exception(f"HuggingFace task:{task} not supported in EVA currently")
+
+    task_input = SUPPORTED_TASKS.get(task, {}).get("type", None)
+    if task_input is None:
+        raise Exception("Task Type Could Not Be Inferred")
+
+    pipe = pipeline(**pipeline_args)
+    if task_input == "text":
```

Comment:
A lot of hard-coded strings. Will making them enums help? What is HF version specific here?